### PR TITLE
Remove test linalg test skips from MAGMA integration

### LINKED ISSUE #55552: ROCM: After enabling Magma, 4 tests are failing in ROCM CI
## 🐛 Bug

As ROCM CI moved to ROCM 4.1, 5k new test cases from test_linalg.py got enabled.
Out of these 4 test cases are failing. These amount to 16 test cases (considering the dtypes).
Need to disable these test cases temporarily for ROCM CI to progress.

## To Reproduce

1. python3.6 test/test_linalg.py -v

## Expected behavior
The test suite should pass

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: Device 6860
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: 3.27.5
MIOpen runtime version: 2.11.0

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] torch==1.9.0a0+gitf5675f8
[pip3] torchvision==0.9.0a0+8a2dc6f
[conda] blas                      1.0                         mkl  
[conda] mkl                       2020.2                      256  
[conda] mkl-include               2020.2                      256  
[conda] mkl-service               2.3.0            py36he8ac12f_0  
[conda] mkl_fft                   1.3.0            py36h54f3939_0  
[conda] mkl_random                1.1.1            py36h0573a6f_0  
[conda] numpy                     1.18.5           py36ha1c710e_0  
[conda] numpy-base                1.18.5           py36hde5b4d6_0  
[conda] torch                     1.9.0a0+gitc691bf2          pypi_0    pypi

## Additional context

cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport

### PR DESCRIPTION
**Pull request by amathews-amd**
Fixes #55552; majority of cases in #51303 

Tests in torch/testing/_internal/common_methods_invocations.py  (tested through test_ops) cannot be fully removed, since the machines seem to be running out of gpu memory during the test, and needs further analysis

### CODE REVIEW

**Comment by amathews-amd** [comment]
Was not able to remove skip from test_lu_unpack, due to failing test from changes in 
https://github.com/pytorch/pytorch/pull/46913

Test passes when run individually (without skipCUDAIfRocm) : 
PYTORCH_TEST_WITH_ROCM=1 python test_linalg.py TestLinalgCUDA.test_lu_unpack_cuda_float32

**Comment by amathews-amd** [comment]
@jithunnair-amd @pruthvistony @jeffdaily @sunway513 - please review

**Comment by amathews-amd** [comment]
test_cond seems to be failing, presumably due to commit https://github.com/pytorch/pytorch/pull/58040 

Error is:
```
22:54:39 ======================================================================
22:54:39 ERROR [0.065s]: test_cond_cuda_complex128 (__main__.TestLinalgCUDA)
22:54:39 ----------------------------------------------------------------------
22:54:39 Traceback (most recent call last):
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1033, in wrapper
22:54:39     method(*args, **kwargs)
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 297, in instantiated_test
22:54:39     raise rte
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
22:54:39     result = test_fn(self, *args)
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 617, in dep_fn
22:54:39     return fn(slf, device, *args, **kwargs)
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 617, in dep_fn
22:54:39     return fn(slf, device, *args, **kwargs)
22:54:39   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 617, in dep_fn
22:54:39     return fn(slf, device, *args, **kwargs)
22:54:39   File "test_linalg.py", line 1576, in test_cond
22:54:39     run_test_case(input, p)
22:54:39   File "test_linalg.py", line 1560, in run_test_case
22:54:39     result = torch.linalg.cond(input, p)
22:54:39 RuntimeError: expected self and mask to be on the same device, but got mask on cpu and self on cuda:0
22:54:39 
```

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu [hunk 1]
```diff
@@ -1425,10 +1425,6 @@ AT_ERROR("inverse: MAGMA library not found in "
   magma_int_t lda = std::max<magma_int_t>(1, n);
   magma_int_t lwork = n * magmaGetriOptimalBlocksize<scalar_t>(n);
 
-  // magmaLu and magmaGetri requires infos tensor to live on CPU
-  infos_lu = infos_lu.to(at::kCPU);
-  infos_getri = infos_getri.to(at::kCPU);
-
```
Comment (jithunnair-amd) [inline aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:1431]: @IvanYashchuk please review this in light of https://github.com/pytorch/pytorch/commit/a49406b33136a8c12503d00b1410e5596e19aede#r50939460

Comment (amathews-amd) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:1431]: Opened an issue to check other parts of code
https://github.com/pytorch/pytorch/issues/58769

Comment (IvanYashchuk) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:1431]: This change is not correct. `infos_lu` and `infos_getri` tensors are created on GPU by default. MAGMA calls wrapped in `magmaLu` and `magmaGetri` expect the info variables on host memory and will segfault with this change (checked with CUDA version < 10.1 and MAGMA 2.5.2). This segfault would be caught only when submitted from `ci-all/` branch, there is a CUDA 9 build IIRC. I've compiled the latest master branch of MAGMA on the ROCm platform and I see that there is no segfault. Apparently, the reason is that it's fine to dereference device pointers in host code on ROCm, but it's illegal on CUDA.
I'll submit an alternative fix that would work both for CUDA and ROCm.

Comment (amathews-amd) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:1431]: Moved this PR to draft mode; Will complete this PR once https://github.com/pytorch/pytorch/pull/59223 is done.

Comment (amathews-amd) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu]: Moving this PR out of draft mode.

**Review by jithunnair-amd — APPROVED** [review]
LGTM

**Comment by jithunnair-amd** [comment]
@malfet This PR enables more magma-related PyTorch unit tests for ROCm. Please review and merge.

**Comment by jeffdaily** [comment]
@malfet now that pytorch 1.9 is out, do you have a chance to revist this PR review?  Thanks.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**